### PR TITLE
docs: Add more details on how URLs are verified

### PR DESCRIPTION
**Pull request by facutuesca**
Finishes work on https://github.com/pypi/warehouse/issues/16486, adding details on which URLs are verified for each specific Trusted Publishing provider, along with examples of all currently supported URLs

/cc @DarkaMaul @woodruffw

### CODE REVIEW

File: docs/user/project_metadata.md [hunk 1]
```diff
@@ -30,21 +30,44 @@ PyPI currently supports several ways of verifying project URLs. When a URL is ve
...
+    - https://github.com/pypa/pip.git
+    - https://pypa.github.io/pip
+    - https://pypa.github.io/pip/* (all subpaths)
+- [GitLab CI/CD][gitlab-tab]: Packages uploaded using GitLab CI/CD from a repository will have the GitLab URLs for that
+  repository verified. For example, for the `pypa/pip` repository, the following URLs will be verified:
+    - https://gitlab.com/pypa/pip
+    - https://gitlab.com/pypa/pip/* (all subpaths)
+    - https://gitlab.com/pypa/pip.git
+- [Google Cloud][gc-tab]: No Google-specific URLs are currently verified.
+- [ActiveState][active-tab]: Packages uploaded using ActiveState will have URLs linking to the project in ActiveState
```
Comment (DarkaMaul) [inline docs/user/project_metadata.md:66]:
```suggestion
- [ActiveState][active-tab]: Packages uploaded using ActiveState will have URLs linked to the project in ActiveState 
```

Comment (DarkaMaul) [reply docs/user/project_metadata.md:66]: NITPICK: Is there any reason not to reuse the `pypi/pip` example for ActiveState?

File: docs/user/project_metadata.md [hunk 2]
```diff
@@ -30,21 +30,44 @@ PyPI currently supports several ways of verifying project URLs. When a URL is ve
...
     URL or any other relationship to the project in question.
 
-### Self-links
+The following subsections specify the different types of URLs that can be verified.
 
-Any URL to the project on PyPI will be considered verified.
+#### Self-links
 
-### Via Trusted Publishing
+Any URL to the project on PyPI will be considered verified. For example, the project page for `pip` will mark all of the
```
Comment (DarkaMaul) [inline docs/user/project_metadata.md:37]:
```suggestion
PyPI considers any project URL pointing on PyPI as verified. For example, the project page for `pip` will mark all of the
```

File: docs/user/project_metadata.md (hunk 1 above)
Comment (facutuesca) [reply docs/user/project_metadata.md:66]: nope, fixed!

File: docs/user/project_metadata.md (hunk 2 above)
Comment (facutuesca) [reply docs/user/project_metadata.md:37]: I think this changes the meaning. Saying "PyPI considers any project URL pointing ~on~ to PyPI as verified` is too broad ("pointing to PyPI" is too generic)

 Maybe we could say "PyPI considers any project URL pointing to that project's PyPI page as verified" ?

File: docs/user/project_metadata.md (hunk 1 above)
Comment (facutuesca) [reply docs/user/project_metadata.md:66]: For the first suggestion, I'm not sure if "having URLs linked to" is better than "having URLs linking to". I'll leave the question open for the native English speakers

Comment (woodruffw) [reply docs/user/project_metadata.md:66]: I think both are idiomatic, but "linked" is probably what I would have written. I'll accept this suggestion 🙂

**Review by di — APPROVED** [review]

File: docs/user/project_metadata.md (hunk 2 above)
Comment (di) [reply docs/user/project_metadata.md:37]: I think this is OK as is.

File: docs/user/project_metadata.md [hunk 3]
```diff
@@ -30,21 +30,44 @@ PyPI currently supports several ways of verifying project URLs. When a URL is ve
...
-PyPI supports the following OIDC Publishers:
+The URLs that can be verified depend on the Trusted Publisher used:
+
+- [GitHub Actions][gh-action-tab]: Packages uploaded using GHA from a repository will have the GitHub URLs for that
+  repository verified. For example, for the `pypa/pip` repository, the following URLs will be verified:
+    - https://github.com/pypa/pip
+    - https://github.com/pypa/pip/* (all subpaths)
+    - https://github.com/pypa/pip.git
+    - https://pypa.github.io/pip
+    - https://pypa.github.io/pip/* (all subpaths)
```
Comment (woodruffw) [inline docs/user/project_metadata.md:59]: Minor thing, but consider putting these (and below) in backticks -- IMO having them formatted as code better clarifies that we're not telling people to click on them.

File: docs/user/project_metadata.md (hunk 2 above)
Comment (woodruffw) [reply docs/user/project_metadata.md:37]: > Maybe we could say "PyPI considers any project URL pointing to that project's PyPI page as verified" ?

That sounds good to me!

File: docs/user/project_metadata.md [hunk 4]
```diff
@@ -30,21 +30,44 @@ PyPI currently supports several ways of verifying project URLs. When a URL is ve
...
 
-### Via Trusted Publishing
+Any URL to the project on PyPI will be considered verified. For example, the project page for `pip` will mark all of the
+following as verified:
+
+- https://pypi.org/project/pip/
+- https://pypi.org/p/pip/
+- https://pypi.python.org/project/pip
+- https://pypi.python.org/p/pip
+- https://python.org/pypi/pip
```
Comment (di) [inline docs/user/project_metadata.md:44]:
```suggestion
- `https://pypi.org/project/pip/`
- `https://pypi.org/p/pip/`
- `https://pypi.python.org/project/pip`
- `https://pypi.python.org/p/pip`
- `https://python.org/pypi/pip`
```

File: docs/user/project_metadata.md (hunk 3 above)
Comment (di) [inline docs/user/project_metadata.md:59]:
```suggestion
    - `https://github.com/pypa/pip`
    - `https://github.com/pypa/pip/*` (all subpaths)
    - `https://github.com/pypa/pip.git`
    - `https://pypa.github.io/pip`
    - `https://pypa.github.io/pip/*` (all subpaths)
```

File: docs/user/project_metadata.md [hunk 5]
```diff
@@ -30,21 +30,44 @@ PyPI currently supports several ways of verifying project URLs. When a URL is ve
...
+    - https://github.com/pypa/pip
+    - https://github.com/pypa/pip/* (all subpaths)
+    - https://github.com/pypa/pip.git
+    - https://pypa.github.io/pip
+    - https://pypa.github.io/pip/* (all subpaths)
+- [GitLab CI/CD][gitlab-tab]: Packages uploaded using GitLab CI/CD from a repository will have the GitLab URLs for that
+  repository verified. For example, for the `pypa/pip` repository, the following URLs will be verified:
+    - https://gitlab.com/pypa/pip
+    - https://gitlab.com/pypa/pip/* (all subpaths)
+    - https://gitlab.com/pypa/pip.git
```
Comment (di) [inline docs/user/project_metadata.md:64]:
```suggestion
    - `https://gitlab.com/pypa/pip`
    - `https://gitlab.com/pypa/pip/*` (all subpaths)
    - `https://gitlab.com/pypa/pip.git`
```

File: docs/user/project_metadata.md [hunk 6]
```diff
@@ -30,21 +30,44 @@ PyPI currently supports several ways of verifying project URLs. When a URL is ve
...
+- [Google Cloud][gc-tab]: No Google-specific URLs are currently verified.
+- [ActiveState][active-tab]: Packages uploaded using ActiveState will have URLs linked to the project in ActiveState 
+  verified:
+    - https://platform.activestate.com/pypa/pip
+    - https://platform.activestate.com/pypa/pip/* (all subpaths)
```
Comment (di) [inline docs/user/project_metadata.md:69]:
```suggestion
    - `https://platform.activestate.com/pypa/pip`
    - `https://platform.activestate.com/pypa/pip/*` (all subpaths)
```

File: docs/user/project_metadata.md (hunk 2 above)
Comment (di) [inline docs/user/project_metadata.md:37]:
```suggestion
PyPI considers any URL pointing to that project on PyPI as verified.
For example, the project page for `pip` will mark all of the
```

**Pull request merged**